snakemake/remote/AzBlob.py

```diff
@@ -226,6 +226,8 @@ def upload_to_azure_storage(
             container_client.create_container()
         except azure.core.exceptions.ResourceExistsError:
             pass
+        except azure.core.exceptions.ClientAuthenticationError:
```

Comment:
This needs an explaining comment. Why is the authentication error not an issue here. To me it means that the container has not been created, which would be a problem, right?

Reply:
A SAS token can be created in the level of a container. Unlike a SAS token that is created in the level of the entire storage account, this SAS token is not allowed to perform the create_container operation. 
So, even though the container exists, and the create_container should return the ResourceExistsError, since by this SAS token it's not allowed to perform the operation, the ClientAuthenticationError error will be raised beforehand. Thus, the whole upload fails. 
Does it clarify the situation?

Reply:
Ok, understood. And it is impossible that the container does not exist and this exception is still raised? If not, maybe one could check for existence in the exception handling?

Reply:
I might miss your answer (I get lots of github notifications), please feel free to contact me via discord.

Reply:
Thanks Johannes for the feedback. I thought of checking for the existence of the container, and only try to create it if it doesn't exist. I guess it would cover your concern to some extent. I cannot think of any scenario where your point will be raised.

It will nevertheless still fail for the scenario that I have described on checking the existence of the container. I cannot find any other alternative than passing this exception. 
I have issued a new pull request as I realized the ClientAuthenticationError is not going to capture the error correctly (looks strange). In the new PR I added the condition for checking whether container exists.